lib/instrumentation/openai.js

```diff
@@ -23,6 +23,64 @@ const { DESTINATIONS } = require('../config/attribute-filter')
 
 let TRACKING_METRIC = OPENAI.TRACKING_PREFIX
 
+/**
+ * Parses the response from OpenAI and extracts the message content and role.
+ *
+ * @param {object} response The OpenAI SDK response object
+ * @returns {{ content: string, role: string }} the message object with fields `content` and `role`
+ */
+function getMessageFromResponse(response) {
+  const message = { content: undefined, role: undefined }
+  if (response?.object === 'response') {
+    message.content = response?.output?.[0]?.content?.[0]?.text
+    message.role = response?.output?.[0]?.role
+  } else {
+    message.content = response?.choices?.[0]?.message?.content
+    message.role = response?.choices?.[0]?.message?.role
+  }
+
+  return message
```

Comment:
I'd think this could be a bit more compact. What about something like this?
```suggestion
  let content
  let role 
  if (response?.object === 'response') {
    content = response?.output?.[0]?.content?.[0]?.text
    role = response?.output?.[0]?.role
  } else {
    content = response?.choices?.[0]?.message?.content
    role = response?.choices?.[0]?.message?.role
  }

  return { content, role }
```